### PR TITLE
Playground: Add support for rangeStart and rangeEnd

### PR DESCRIPTION
**Pull request by JamesHenry**
I was delighted to see that this appears to be very easy to add and would allow users to more easily reproduce and report issues such as:

- https://github.com/prettier/prettier/issues/3789
- https://github.com/prettier/prettier/issues/4206
...etc...

screen shot 2018-03-28 at 00 52 17

### CODE REVIEW

**Comment by j-f1** [comment]
Looks cool 👍 ([live preview](https://deploy-preview-4216--prettier.netlify.com/playground/))

**Comment by JamesHenry** [comment]
And full for reference, here is me recreating the issue I outline here (https://github.com/prettier/prettier/issues/3789#issuecomment-366273671) in the running local version of this PR branch: 

Reproduction in the updated playground from this PR:
screen shot 2018-03-28 at 01 04 01

**Comment by JamesHenry** [comment]
Ah nice, yeah thanks @j-f1!

Here is the same reproduction on the live preview as an example: https://deploy-preview-4216--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACMAGwEMBnUvAJQFdC49gAdKPA6UmbasGHACmatWEDDACW7JELwBJAPKiJUUgBpBeAJQMAvs20gVIEePbJQxbNggB3AAoWEpZCGIA3CGIAmBlx2QAzYkJSOEMAI2xiMABrOBgAZQwosSgAc2ROalCQTwgwAKCQwxSQ3FtI1IBbYgLg7IArUjQAIUiYuPjiSrgAGRS4WqKjahgMEYAmQeyk7FLnGABPDDhSMGwxUR8MddgAdS8YAAtkAA4ABkNtiBDdyIxnbZW4bFcBw0i0uABRKG8kADYLiAPqk4PEYBYYMgAMwAFnecAAjtQxNg4OViFUakhAnVDCFKmIMlxsqQUqk6ABFagQeBTQwQsL7TxHZDjBmRMSEckAYQglWqzig0DeIGoIQAKsQwk4cYU4NptEA

File: website/static/playground.js [hunk 1]
```diff
@@ -194,6 +196,9 @@ window.onload = function() {
 
 function setOptions(options) {
   OPTIONS.forEach(function(option) {
+    if (option === undefined) {
+      return;
+    }
```
Comment (suchipi) [inline website/static/playground.js]: What's this check for? We're iterating over an array of known values, so there shouldn't be any undefined values here

File: website/pages/playground/index.html [hunk 2]
```diff
@@ -95,6 +95,8 @@ <h1>Prettier <span id="version"></span></h1>
           <label>--print-width <input type="number" value="80" min="0" id="printWidth"></label>
           <label>--tab-width <input type="number" min="0" value="2" id="tabWidth"></label>
           <label><input type="checkbox" id="useTabs"> --use-tabs</label>
+          <label>--range-start <input type="number" min="0" id="rangeStart"></label>
+          <label>--range-end <input type="number" min="0" id="rangeEnd"></label>
```
Comment (ikatyang) [inline website/pages/playground/index.html]: I guess these should belong to `Special` category.

**Comment by lydell** [comment]
Just an idea: What if you could set the range by selecting a piece of text somehow? Probably another PR though. Possibly not even useful.

**Comment by suchipi** [comment]
I like that idea. If that proves difficult, we could render the start/end markers as lines behind the text.

**Comment by JamesHenry** [comment]
Sorry for the delay on looping back to this, I finally found a few mins to look into CodeMirror.

I have hacked together a working version of the overlay API that CodeMirror provides to perform highlighting of the range. Pretty darn cool! 😄 

Here is the same example as the one above but with my local updates:
screen shot 2018-04-04 at 10 29 21

Happy for me to push that up as a part of this PR?

**Comment by josephfrazier** [comment]
Oh, if you've already got it working, by all means include it! That would make it much easier to know what the range actually is and to get it right.

FWIW, I'm also curious about https://github.com/prettier/prettier/pull/4216#discussion_r177611888

**Comment by JamesHenry** [comment]
I'm afraid I have no idea about the undefined check 🤷‍♂️ will remove it.

Here is my suggestion for the organisation, now that we have a bit more behaviour (the highlighting), I think it is worth having its own section:

screen shot 2018-04-04 at 11 58 28

**Comment by JamesHenry** [comment]
Same example from above but with all of the new updates:

https://deploy-preview-4216--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACMAGwEMBnUvAJQFdC49gAdKPA6UmbasGHACmatWEDDACW7JELwBJAPKiJUUgBpBeAJQMAvs20gVIEePbJQGbGNgB1MQBMYAC2QAOAAyGYxAEa2HzpAAmQ1IrAHM6AEVqCHhkADNiQlI4T2xiMUJwgGEIAFs84mQQKGg4AxBvdLAAazgYAGUMYjBw5E5qVJAAK1I0ACFqusbiPLgAGStypETkrubsFOximABPDDhSMEtRCpS8sXauLuoUgBUfUgSklMMrJZgABXSwwuu5w2w4AEdqMS-nsRXkUZjd5tgIClrOkMMULJs4NgAG7lQzEbAQgDuj3RCCuKGISIg9gq6SgYTgDS8uGQAGYACyfYjkuAAUSgdmQADYPCA7BAwO9biAyDAhV0INQYBgpYFxdptEA

Ready for final review!

File: website/pages/playground/index.html [hunk 3]
```diff
@@ -20,25 +20,25 @@
...
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/mode/css/css.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/mode/markdown/markdown.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/addon/display/placeholder.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/addon/display/rulers.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/addon/search/searchcursor.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/addon/edit/matchbrackets.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/addon/edit/closebrackets.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/addon/comment/comment.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/addon/wrap/hardwrap.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.26.0/keymap/sublime.js"></script>
```
Comment (lydell) [inline website/pages/playground/index.html]: You need to update all of these in website/static/service-worker.js as well

Comment (JamesHenry) [reply website/pages/playground/index.html]: Done

**Pull request merged**